Test/pruning/NonmonomorphicSplit.bpl

```diff
@@ -0,0 +1,37 @@
+// RUN: %parallel-boogie /prune /trace /errorTrace:0 "%s" > "%t"
+// RUN: %OutputCheck "%s" --file-to-check="%t"
+
+// Related PR #767.
+
+function f1 (x: int) : int;
+function f2 (x: int) : int uses
+{
+  axiom f1(0) == 1 && f2(0) == 2;
+}
+
+// Above axiom will be split into two during monomorphization into
+// axiom f1(0) == 1 and axiom f2(0) == 2.
+// Current implementation ensures that f1(0) == 1 is kept as a dependency
+// of f1, and f2(0) is moved to be a dependency of f2. If there would be
+// other symbols in the axiom, the new split axioms would be added to those
```

Comment:
I sort of understand what the last sentence is saying. But I am confident its meaning can be made more transparent. It is fine to define symbols and use them later to clarify what is being said.

Reply:
Thanks for the feedback, they were written a bit hastily... I tried updating both explanations based on the comments.